library/common/src/main/java/com/epmedu/animeal/common/component/GpsSettingsProvider.kt

```diff
@@ -0,0 +1,14 @@
+package com.epmedu.animeal.common.component
+
+import kotlinx.coroutines.flow.Flow
+
+/** Provides info about GPS setting */
+interface GpsSettingsProvider {
+
+    val isGpsSettingsEnabled: Boolean
+
+    /** Listener of Gps setting events (enabled(true)/disabled(false)) */
+    fun fetchGpsSettingUpdates(): Flow<Boolean>
```

Comment:
can be just `gpsSettingUpdates(): Flow<Boolean>`

Also, you have added comment to clarify whats going on. To avoid misunderstanding we can add `sealed interface` with different states